receiver/windowsservicereceiver/winservice.go

```diff
@@ -1,39 +1,68 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
-
-//revive:disable:unused-parameter
 //go:build windows
 
-package windowsservicereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsservicereceiver"
-
-import "golang.org/x/sys/windows/svc/mgr"
+package windowsservicereceiver
 
-/**
-* Windows Service representation and associated functions. These handle
-* interacting with the SCM and martialing service information returned by the
-* windows api calls.
-**/
+import (
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
 
-// receiver representation of a service
-//
-//nolint:unused
 type winService struct {
-	service       *mgr.Service
-	serviceStatus uint32
-	startType     uint32
+	mgr    *serviceManager
+	name   string
+	handle *mgr.Service
+
+	status svc.Status
+	config ConfigEx
 }
 
-//nolint:unused
-func getService(mgr *serviceManager, sname string) (*winService, error) {
-	return &winService{}, nil
+func getService(m *serviceManager, sname string) (*winService, error) {
+	m.targetService = sname
+	s, err := m.openService()
+	if err != nil {
+		return nil, err
+	}
+	return &winService{
+		mgr:    m,
+		name:   sname,
+		handle: s,
+	}, nil
 }
 
-//nolint:unused
-func (*winService) getStatus() error {
+func (ws *winService) getStatus() error {
```

Comment:
To me `getStatus` sounds like is returning status and possibly an error, like `getService` - perhaps a different name like `updateStatus` will make the difference clear. Same thing to `getConfig` below.